crates/ruma-client-api/src/error.rs

```diff
@@ -135,9 +137,13 @@ pub enum ErrorKind {
     CannotLeaveServerNoticeRoom,
 
     #[doc(hidden)]
-    _Custom { errcode: String, extra: BTreeMap<String, JsonValue> },
+    _Custom { errcode: PrivOwnedStr, extra: Extra },
```

Comment:
(This counts for all changes)

The only concern I have is read accessibility, I'm seeing Extra in some spots, would these still be reachable for the consumer to extract.

Also, just so we're on the same page;  this is fine as long as it makes directly constructing these variants via enum reference impossible, however, if it makes it impossible to construct them *at all* from the library consumer side, then I don't think this change is a good idea. 

As long as there is an idiomatic way for them to construct this, like Into, for all variants here, and there's a way to extract that data again, then this would be fine.

Reply:
> The only concern I have is read accessibility, I'm seeing Extra in some spots, would these still be reachable for the consumer to extract.

This PR changes no parts of the "truly" public API (public and documented). Everything that is `#[doc(hidden)]` was already things you should never be using, just without the compiler enforcing it.